pkg/api/customization/project/project_store.go

```diff
@@ -189,6 +232,47 @@ func isQuotaFit(apiContext *types.APIContext, nsQuotaLimit *v3.ResourceQuotaLimi
 			msg))
 	}
 
+	if len(limitToAdd) == 0 && len(limitToRemove) == 0 {
+		return nil
+	}
+
+	// check if default quota is enough to set on namespaces
+	mu := resourcequota.GetProjectLock(id)
+	mu.Lock()
+	defer mu.Unlock()
+	var namespaces []clusterclient.Namespace
+	options := &types.QueryOptions{
+		Conditions: []*types.QueryCondition{
+			types.NewConditionFromString("projectId", types.ModifierEQ, id),
+		},
+	}
+	if err := access.List(apiContext, &clusterchema.Version, clusterclient.NamespaceType, options, &namespaces); err != nil {
+		return err
+	}
+
+	var nsLimits []*v3.ResourceQuotaLimit
+	for i := 0; i < len(namespaces); i++ {
+		toAppend := &mgmtclient.ResourceQuotaLimit{}
```

Comment:
I think lines 255-262 can be done only once outside the loop, since limitToAdd won't change

Reply:
@kinarashah that's reasonable, will change